pkg/deployment/baremetal.go

```diff
@@ -49,7 +49,7 @@ func DeployBaremetalSet(
 		instance.Spec.BaremetalSetTemplate.DeepCopyInto(&baremetalSet.Spec)
 		for _, node := range nodes.Items {
 			ipSet, ok := ipSets[node.Name]
-			instanceSpec := baremetalSet.Spec.BaremetalHosts[node.Name]
+			instanceSpec := baremetalSet.Spec.BaremetalHosts[node.Spec.HostName]
```

Comment:
I am not into the details of the dataplane-operator. The HostName is an optional parameter [1], should it be required now?

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/openstackdataplanenode_types.go#L32

Reply:
oops.. I thot is is required field, but I see at many places we do[1]. Though we can do it in the webhook defaulting, I'll add the same here for the time being.

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/pkg/deployment/ipam.go#L78C1-L82C5